app/User.php

```diff
@@ -73,6 +76,11 @@ public function role()
         return $this->hasOne('App\Role', 'idroles', 'role');
     }
 
+    public function repairdir_role() {
+        // Make sure we don't return a null value.  The client select would struggle with null values.
+        return $this->repairdir_role ? $this->repairdir_role : Role::REPAIR_DIRECTORY_NONE;
```

Comment:
Ah looks like it does prefer 0 to null then - we'll have to figure a way around this.  (Either we make Repair Directory handle 0 as a valid role id, or we make the select work with null values.)